### ISSUE TITLE
Project Custom Attribute returns null

### ISSUE THREAD

**Issue by abhikum1** [gitlab4j/gitlab4j-api#579]
Gitlab Version Tested on: 11.10.8 and 12.8
Gitlab4j version: 4.14.30

Issue: Trying to get the custom attributes of a project using the project api.

Method:

`List<CustomAttribute> cAttrs = gitLabApi.getProjectApi().getProject(104).getCustomAttributes();`

The value returned is null.
i was able to successfully  retrieve the values from the rest api via the rest client.
![image](https://user-images.githubusercontent.com/53219459/82909963-9f493d80-9f87-11ea-90df-4e06d27be04e.png)

**Comment by abhikum1** [comment]
Update:
The method getCustomAttributes is able to retrieve the results when using an admin token, however if using a token of a normal user who just has a developer access results in the null pointer exception.

**Comment by gmessner** [comment]
@abhikum1 In order to retrieve the project with the custom attributes populated you must call the folowing method on the ```ProjectApi``` and set the ```withCustomAttributes```param to ```true``` (you can pass null or false for ```includeStatistics``` and ```includeLicense``` if you don't want that info):
```
public Project getProject(Object projectIdOrPath, Boolean includeStatistics,
	    Boolean includeLicense, Boolean withCustomAttributes)
```

**Comment by abhikum1** [comment]
Hello @gmessner ,

Thanks for this information it works. so my use case is to modify the custom attributes values in bulk for all the projects inside gitlab.

i have written the following test code:
```
GitLabApi gitLabApi = new GitLabApi("https://servername/git", "JGtfR_aWq3kLwi2v-rGn");
		Project project = gitLabApi.getProjectApi().getProject(176, Boolean.FALSE, Boolean.FALSE, Boolean.TRUE);
		List<CustomAttribute> attrs  = project.getCustomAttributes();
		for (CustomAttribute customAttribute : attrs) {
			if(customAttribute.getKey().equalsIgnoreCase("rtc_repourl")) {
				customAttribute.setValue(customAttribute.getValue().replace("somesourcevalue", "somedestvalue"));
			}
		}
		gitLabApi.getProjectApi().updateProject(project);

	}
```
On enabling logging:

This is the response i get from the post request:
```
May 27, 2020 12:56:22 PM org.gitlab4j.api.utils.MaskingLoggingFilter log
INFO: 2 - Sending client request on thread main
2 > PUT https://servername/git/api/v4/projects/176
2 > Accept: application/json
2 > Content-Type: application/x-www-form-urlencoded
2 > PRIVATE-TOKEN: ********

May 27, 2020 12:56:24 PM org.gitlab4j.api.utils.MaskingLoggingFilter log
INFO: 2 - Received server response on thread main
2 < 200
2 < Cache-Control: max-age=0, private, must-revalidate
2 < Connection: Keep-Alive
2 < Content-Length: 2628
2 < Content-Type: application/json
2 < Date: Wed, 27 May 2020 07:25:20 GMT
2 < Etag: W/"f3c5b22df73fcfa6cfe2c66f36dc6b3f"
2 < Keep-Alive: timeout=5, max=100
2 < Referrer-Policy: strict-origin-when-cross-origin
2 < Strict-Transport-Security: max-age=31536000
2 < Vary: Origin
2 < X-Content-Type-Options: nosniff
2 < X-Frame-Options: SAMEORIGIN
2 < X-Request-Id: M96z2sKuSI8
2 < X-Runtime: 0.477181
```
The Response Body of the project class:
```
{
  "approvalsBeforeMerge" : null,
  "archived" : false,
  "avatarUrl" : null,
  "containerRegistryEnabled" : true,
  "createdAt" : "2020-05-02T14:24:01.173Z",
  "creatorId" : 1,
  "defaultBranch" : "master",
  "description" : null,
  "forksCount" : 0,
  "forkedFromProject" : null,
  "httpUrlToRepo" : "https://serversname/git/projectname/sw_vvd2_app.git",
  "id" : 176,
  "issuesEnabled" : false,
  "jobsEnabled" : true,
  "lastActivityAt" : "2020-05-02T14:24:01.173Z",
  "lfsEnabled" : true,
  "mergeMethod" : "merge",
  "mergeRequestsEnabled" : true,
  "name" : "SW_VVD2_App",
  "namespace" : {
    "id" : 118,
    "name" : "projectname",
    "path" : "projectname",
    "kind" : "group",
    "fullPath" : "projectname"
  },
  "nameWithNamespace" : "projectname / SW_VVD2_App",
  "onlyAllowMergeIfPipelineSucceeds" : false,
  "onlyAllowMergeIfAllDiscussionsAreResolved" : false,
  "openIssuesCount" : null,
  "owner" : null,
  "path" : "sw_vvd2_app",
  "pathWithNamespace" : "projectname/sw_vvd2_app",
  "permissions" : {
    "projectAccess" : null,
    "groupAccess" : {
      "accessLevel" : 50,
      "notificationLevel" : 3
    }
  },
  "publicJobs" : false,
  "repositoryStorage" : "default",
  "requestAccessEnabled" : false,
  "runnersToken" : "rMvqhMEM1igGsrX3Fiza",
  "sharedRunnersEnabled" : true,
  "sharedWithGroups" : [ ],
  "snippetsEnabled" : true,
  "sshUrlToRepo" : "git@servername:projectname/sw_vvd2_app.git",
  "starCount" : 0,
  "tagList" : [ "configuration" ],
  "visibilityLevel" : null,
  "visibility" : "internal",
  "wallEnabled" : null,
  "webUrl" : "https://servername/git/iMUX800/sw_vvd2_app",
  "wikiEnabled" : true,
  "printingMergeRequestLinkEnabled" : false,
  "resolveOutdatedDiffDiscussions" : false,
  "statistics" : null,
  "initializeWithReadme" : null,
  "packagesEnabled" : null,
  "emptyRepo" : false,
  "licenseUrl" : null,
  "license" : null,
  "customAttributes" : [ {
    "key" : "alm",
    "value" : "script"
  }, {
    "key" : "hooks",
    "value" : "512000"
  }, {
    "key" : "rtc_repokey",
    "value" : "6e9504896ca64f209c04efee31192f10"
  }, {
    "key" : "rtc_repourl",
    "value" : "https://replacedtext/ccm"
  } ],
  "markedForDeletionOn" : null,
  "public" : null
}
```
i see that the text is replaced successfully in the "rtc_repourl" custom attr and the response code of this method 200, but the custom attributes are not updated.
Any Ideas? or is it the case that update project method is not meant to update the custom attribute?

Thanks in advance.

**Comment by gmessner** [comment]
@abhikum1  My apologies for the late response, but I have been hospitalized with a major accident, and am now just out of the hospital.

The GitLab update project REST API does not update custom attributes, it is a separate API call which unfortunately is not implemented in this library yet. I expect to add support later this week and release on 6/27.

**Comment by gmessner** [comment]
@abhikum1 Full support for the Project Custom Attributes API is now available in release 4.15.1.  Please verify and close this issue when you have time.

You'll find the following methods in ```ProjectApi```:
```
public List<CustomAttribute> getCustomAttributes(final Object projectIdOrPath)
public Pager<CustomAttribute> getCustomAttributes(final Object projectIdOrPath, int itemsPerPage)
public Stream<CustomAttribute> getCustomAttributesStream(final Object projectIdOrPath)
public CustomAttribute getCustomAttribute(final Object projectIdOrPath, final String key)
public Optional<CustomAttribute> geOptionalCustomAttribute(final Object projectIdOrPath, final String key)
public CustomAttribute setCustomAttribute(final Object projectIdOrPath, final String key, final String value)
public void deleteCustomAttribute(final Object projectIdOrPath, final String key)
```

**Comment by gmessner** [comment]
@abhikum1 Since this functionality was released a week ago I'm closing this issue. If you have any problems with it please let me know.
